### PR TITLE
Narrative Filter

### PR DESCRIPTION
**Pull request by AdamZarger**
PR'ing the changes in progress against my own repo in order to get feedback/early eyes-on.

Creating a SingleCheckbox filter for the `has_narrative` parameters.

## Additions

- SingleCheckbox filter
- Logic to handle a boolean/flag parameter (query string, elasticsearch call, etc)
- Fixed all broken unit tests

## Review

- @user

## Todos

- Write tests to get coverage up

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by AdamZarger** [comment]
Current Status:
* SingleCheckbox for "Only Show Complaints With Narratives" will now properly pre-check itself if "has_narratives" are in the query string
* Functionality for the on-click event for the check box does not work. It appears to fire, but does not update the page with `filterChanged()`

File: src/Filters/SingleCheckbox.jsx [hunk 1]
```diff
@@ -1,15 +1,75 @@
...
+SingleCheckbox.defaultProps = {
+  isChecked: false
+}
+
+export const mapStateToProps = state => ( {
+  isChecked: typeof state.query.has_narrative !== 'undefined' &&
+    ( state.query.has_narrative.toString() === 'yes' ||
+      state.query.has_narrative.toString() === 'true' )
+} )
+
```
Comment (JeffreyMFarley) [inline src/Filters/SingleCheckbox.jsx]: It's usually a good idea to write long chains of accessors into a local variable if you use them more than once:

👎 
```javascript
( state.query.has_narrative.toString() === 'yes' || state.query.has_narrative.toString() === 'true' )
```
👍 
```javascript
const value =  state.query.has_narrative.toString()
( value === 'yes' || value === 'true' )
```

File: src/actions/complaints.jsx [hunk 2]
```diff
@@ -35,6 +35,12 @@ export function stateToQS( state ) {
       value = shortIsoFormat( value )
     }
 
+    // Process boolean flags
+    const positives = [ 'yes', 'true' ]
+    if ( types.flagFilters.indexOf( field ) !== -1 ) {
+      value = positives.includes( String( value ).toLowerCase() )
+    }
```
Comment (JeffreyMFarley) [inline src/actions/complaints.jsx:42]: 👍

File: src/Filters/SingleCheckbox.jsx (hunk 1 above)
Comment (AdamZarger) [reply src/Filters/SingleCheckbox.jsx]: I tried to implement this suggestion, but I wasn't entirely sure how to create a local variable within `mapStateToProps`. It keeps throwing errors about unexpected syntax and reserved keywords

Comment (JeffreyMFarley) [reply src/Filters/SingleCheckbox.jsx]: You have to make it a proper function and not the shorthand return.  Look at other mapStateToProps for examples

**Comment by JeffreyMFarley** [comment]
Also, can you change the target of this merge to `ccdb5-ui/master`? That should help with the extra files

**Comment by AdamZarger** [comment]
Merging this into the master branch of this repo, then opening another PR to merge into cfpb/ccdb5-ui

**Pull request merged**